### PR TITLE
mod_clamav: Scan for viruses in uploaded files with ClamAV

### PR DESCRIPTION
**Pull request by mworrell**
### Description

Add `mod_clamav` to enable virus scanning of uploaded files.

This implementation is a first rough implementation. @ddeboer and I agreed to have a fresh look for the 1.0 where we could add sanitization using quarantine states on the medium record.

Also use the HTML sanitizer on uploaded html files.
(Which people rarely have permission to upload anyway).

### Checklist

- [x] documentation updated
- [ ] tests added
- [x] no BC breaks

/cc @mmzeeman

### CODE REVIEW

**Comment by mworrell** [comment]
@ddeboer This is ready for merge.

File: doc/ref/modules/mod_clamav.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+The following configs are available:
+
+``clamav_ip``
+    IP address of ``clamd``, default to ``"127.0.0.1"``
+
+``clamav_port``
+    Port of ``clamd``, default to ``3310``
+
+``clamav_max_size``
+    The **StreamMaxLength** of ``clamd``, default to ``26214400`` (25M)
```
Comment (ddeboer) [inline doc/ref/modules/mod_clamav.rst]: Is this equal to clamd’s default?

File: modules/mod_clamav/docs/notes.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    SCAN /home/marc/virus/eicar_com.zip
+    /home/marc/virus/eicar_com.zip: Eicar-Test-Signature FOUND
+    Connection closed by foreign host.
+
+Scan an ok file:
+
+    marc@nuc:~$ telnet localhost 3310
+    SCAN /home/marc/PDf-THAT_WILL_NOT_UPLOAD.pdf
+    /home/marc/PDf-THAT_WILL_NOT_UPLOAD.pdf: OK
+    Connection closed by foreign host.
```
Comment (ddeboer) [inline modules/mod_clamav/docs/notes.md]: Should this be committed?

Comment (mworrell) [reply modules/mod_clamav/docs/notes.md]: Just notes how to use, good to keep them around.

File: doc/ref/modules/mod_clamav.rst (hunk 1 above)
Comment (mworrell) [reply doc/ref/modules/mod_clamav.rst]: yes, the default is 25M

**Pull request merged**